examples/watch/timeout-settings.md

```diff
@@ -0,0 +1,68 @@
+
+**This documentation briefly provides information about the `server side` & `client side` connection timeout settings, in the watch request handler.**
+
+---
+
+There are two inputs available in the client, that could be used to set connection timeouts:
+
+- `timeout_seconds`
+- `_request_timeout`
+
+---
+
+#### Sever-side timeout (`kwargs['timeout_seconds'] = n`)
+
+- The value of the argument `timeout_seconds`, **n**, (which is time duration in seconds) is consumed at the server side. It is included in the request URL to the server. 
+  
+  *For eg.* ~ `https://localhost:6443/api/v1/namespaces/default/pods?labelSelector=app%3Ddemo&timeoutSeconds=100&watch=True`
+
+- In case, if the `timeout_seconds` value is set, the value `n` would determine the server-side connection timeout duration.
+
+  *For eg.* ~ if `kwargs['timeout_seconds'] = 3600`, then the server-side connection timeout will be equal to 1 hour.
+  
+  This timeout duration is determined by the expression ~ `timeout = time.Duration(3600) * time.seconds`, *i.e.* `timeout = 1 hour`
+
+  ***Refer:*** 
+  - *[kubernetes/apiserver/pkg/endpoints/handlers/get.go#L255](https://github.com/kubernetes/apiserver/blob/92392ef22153d75b3645b0ae339f89c12767fb52/pkg/endpoints/handlers/get.go#L255)*
+
+- In case, if the `timeout_seconds` value is not set, then the connection timeout will be a randomized value (in seconds) between 0 and `minRequestTimeout`, to spread out load.
+
+  It is determined using the expression ~ `timeout = time.Duration(float64(minRequestTimeout) * (rand.Float64() + 1.0))`
+
+  Where `minRequestTimeout` indicates the minimum number of seconds a handler must keep a request open before timing it out.
+  
+  The default value of `minRequestTimeout` is 1800 seconds.
+
+  ***Refer:***
+  - *[kubernetes/apiserver/pkg/endpoints/handlers/get.go#L258](https://github.com/kubernetes/apiserver/blob/92392ef22153d75b3645b0ae339f89c12767fb52/pkg/endpoints/handlers/get.go#L258)*
+  - *[kubernetes/release-1.1/docs/admin/kube-apiserver.md](https://github.com/kubernetes/kubernetes/blob/release-1.1/docs/admin/kube-apiserver.md)*
+
+- In case of a network outage, this timeout value will have no effect & the client will hang indefinitely without raising any exception.
```

Comment:
what kind of network outage would cause this? it hangs when not accounting for the client side timeout described below, right?

Reply:
Re: 
> what kind of network outage would cause this?

I'm not sure how to define or specify the  **what kind of network outage** part. Could you please phrase it or point me somewhere that talks more about the kind of network outage, I'll read more & try to add a proper definition thee.

Thank you! 

Re:
> It hangs when not accounting for the client side timeout described below, right?

I might be wrong with my understanding, but so far, I understand that it is the server side timeout (i.e. value of argument `timeout_seconds`, that would have no effect in case of the said network outage)

Just a pointer from this comment here ~ https://github.com/kubernetes-client/python/issues/1148#issuecomment-626184613

Reply:
@yliaog, no hurries at all. Just pointing in case, it was missed!

could you please take a look at this review comment too, as what more could be done to improve this section here?

Thank you so much!

Reply:
i think probably add the condition that client side timeout is not specified, something like below:

this timeout value will have no effect & the client will hang indefinitely without raising any exception if client side timeout (see below) is not specified

Reply:
@yliaog, Thank you for the pointers.

I'm rephrasing it as following: ~

> In case of a network outage, the server side timeout value will have no effect & the client will hang indefinitely without raising any exception. Note, that this is the case provided when there is no other client-side timeout (i.e., `_request_timeout`) value specified.
>
> See the section below for information on `client side timeout`